examples/cfd/flow_past_sphere_3d.py

```diff
@@ -78,22 +80,63 @@ def setup_boundary_conditions(self):
         bc_sphere = HalfwayBounceBackBC(indices=sphere)
         self.boundary_conditions = [bc_walls, bc_left, bc_outlet, bc_sphere]
 
-    def setup_boundary_masker(self):
```

Comment:
They can still do that. They're returned.

Reply:
but bc_mask and missing_mask is already changed in a way that's obscured for the user.

Reply:
I don't think that's correct. By defintion, BCs should not overlap (you can't have to BCs in one node on the same bc_mask). So you safely change bc_mask before or after passing it to the solver.


Another option is to create a function like init_stepper in the very base class and overwrite it and pass the data and get them back, and return an error if the stepper is not initialized by the data (this way we can avoid returning everything using __new__).

Reply:
The main issue here is that we’re working with immutable objects. In both backends, this requires handling these objects and ensuring direct access to the kernel data in the examples, especially if one intends to leverage any differentiable capabilities. Naturally, the code might not look exactly like traditional Python NumPy code, but in my opinion, that’s perfectly fine.

Reply:
We can also move the fields creation and initialization (passing the initialization kernel) to the init_stepper. So the fields you get back are already set.

Reply:
so essentially it will look like this:

``
 self.stepper = IncompressibleNavierStokesStepper(omega, boundary_conditions=self.boundary_conditions, collision_type="BGK")
 
self.f_0, self.f_1, self.bc_mask, self.missing_mask =  self.stepper.init()
``
@hsalehipour what do you think about this? (this is the change I wanted to I mentioned)

Reply:
You can safely update missing_mask or bc_mask or the fields after calling init (since you haven't called the stepper itself yet).

Reply:
And we can remove all the helpers to create the fields.

Reply:
Oh and for initialization of the field we can do:

```
self.f_0, self.f_1, self.bc_mask, self.missing_mask =  self.stepper.init(InitializeKernel())
```

Reply:
Another nice thing is that you can just not run init, and create the fields the traditional way more explicitly!